src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java

```diff
@@ -0,0 +1,99 @@
+    private static final String SUCCESSFUL_UPDATE_WITH_EMAIL = SUCCESSFUL_UPDATE + " and email sent";
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult(Const.StatusMessages.STUDENT_NOT_FOUND_FOR_EDIT, HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentUpdateRequest updateRequest = getAndValidateRequestBody(StudentUpdateRequest.class);
+        boolean emailSent = false;
+
+        try {
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.validateTeams(Arrays.asList(student), student.course);
+            logic.updateStudentCascade(
+                    StudentAttributes.updateOptionsBuilder(courseId, studentEmail)
+                            .withName(updateRequest.getName())
+                            .withNewEmail(updateRequest.getEmail())
+                            .withTeamName(updateRequest.getTeam())
+                            .withSectionName(updateRequest.getSection())
+                            .withComment(updateRequest.getComments())
+                            .build());
+
+            if (!student.email.equals(updateRequest.getEmail())) {
+                logic.resetStudentGoogleId(updateRequest.getEmail(), courseId);
+
+                if (updateRequest.getIsSessionSummarySendEmail()) {
+                    emailSent = sendEmail(courseId, updateRequest.getEmail());
+                }
+            }
+        } catch (EnrollException | InvalidParametersException e) {
+            return new JsonResult(e.getMessage(), HttpStatus.SC_BAD_REQUEST);
+        } catch (EntityDoesNotExistException ednee) {
+            return new JsonResult(ednee.getMessage(), HttpStatus.SC_NOT_FOUND);
+        } catch (EntityAlreadyExistsException e) {
+            return new JsonResult("Trying to update to an email that is already in use",
+                    HttpStatus.SC_CONFLICT);
+        }
+
+        return emailSent ? new JsonResult(SUCCESSFUL_UPDATE_WITH_EMAIL) : new JsonResult(SUCCESSFUL_UPDATE);
```

Comment:
I'm not a fan of the ternary. Let's do an early return instead so that it's easier to extend and modify in the future if the need ever arises.

```
if (emailSent) {
    return new JsonResult(...);
}
return new JsonResult(.....);
```

Reply:
Yup updated this part!

@wkurniawan07 agreed! `emailSent == false` does mean those (2) things. However, should we update the message/status we send back to the user based on this fact?

Reply:
> However, should we update the message/status we send back to the user based on this fact?

Ideally.